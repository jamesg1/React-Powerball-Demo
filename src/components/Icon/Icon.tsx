import React from 'react';
import Lightning from './Lightning';
import Trash from './Trash';
import Cross from './Cross';

export interface IconProps {
  width?: number | string;
  fill?: string;
  name: 'lightning' | 'trash' | 'cross';
}

const Icon: React.FC<IconProps> = (props: IconProps) => {
  switch (props.name) {
    case 'lightning':
      return <Lightning {...props} />;
    case 'trash':
      return <Trash {...props} />;
    case 'cross':
      return <Cross {...props} />;
    default:
      return null;
  }
};

Icon.defaultProps = {
  width: '100%',
  fill: '#FFF'
};

export default Icon;
