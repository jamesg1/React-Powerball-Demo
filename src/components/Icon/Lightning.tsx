import React from 'react';
import { IconProps } from './Icon';

const Lightning: React.FC<IconProps> = ({ width, fill }: IconProps) => (
  <svg
    width={width}
    height={width}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink">
    {/* Source: https://iconmonstr.com/connection-1-svg/ */}
    <path fill={fill} d="M8 24l3-9h-9l14-15-3 9h9l-14 15z" />
  </svg>
);

export default Lightning;
