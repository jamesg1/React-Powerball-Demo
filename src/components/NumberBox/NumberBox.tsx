import React from 'react';
import Icon from '../Icon';
import { Theme } from '../../styling';
import { Box } from './Styled';

export interface Props {
  selected?: boolean;
  number: number;
}

const NumberBox: React.FC<Props> = ({ selected, number }: Props) => (
  <Box selected={selected} number={number}>
    {selected && <Icon name="cross" fill={Theme.colors.lightPurple} />}
    <span>{number}</span>
  </Box>
);

export default NumberBox;
