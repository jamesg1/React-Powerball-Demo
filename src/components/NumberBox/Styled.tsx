import styled from 'styled-components';
import { Props } from './NumberBox';

const Box = styled.div<Props>`
  position: relative;
  color: ${props => props.theme.colors.purple};
  font-weight: ${props => props.theme.fontWeights.bold};
  background: ${props =>
    props.selected ? props.theme.colors.lighterPurple : props.theme.colors.white};
  padding: ${props => props.theme.spacing.small};
  width: 10%;
  height: 10%;
  display: flex;
  align-items: center;
  justify-content: center;
  transition: ${props => props.theme.transitions.all}
  border: 1px solid ${props => props.theme.colors.grey};
  margin: 0 -1px -1px 0;
  transition: ${props => props.theme.transitions.all};

  > span {
    z-index: 5;
  }

  > svg {
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    margin: auto;
    width: 75%;
    height: 75%;
    position: absolute;
    fill: white;
  }
`;
export { Box };
