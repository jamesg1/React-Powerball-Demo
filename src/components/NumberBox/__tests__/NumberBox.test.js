import React from 'react';
import NumberBox from '../NumberBox';
import Icon from '../../Icon';
import renderWithTheme from 'styling/withTheme';
import 'jest-styled-components';

describe('<NumberBox />', () => {
  test('it renders when not selected', () => {
    const tree = renderWithTheme(<NumberBox number={25} />);
    const testInstance = tree.root;
    expect(testInstance.findByType('span').props.children).toBe(25);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('it renders cross icon when number is selected', () => {
    const tree = renderWithTheme(<NumberBox number={25} selected={true} />);
    const testInstance = tree.root;
    expect(testInstance.findByType(Icon).props.name).toBe('cross');
    expect(testInstance.findByType('span').props.children).toBe(25);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
