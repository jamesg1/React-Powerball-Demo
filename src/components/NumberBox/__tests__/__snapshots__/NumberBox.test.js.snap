// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberBox /> it renders cross icon when number is selected 1`] = `
.c0 {
  position: relative;
  color: #6C4796;
  font-weight: 700;
  background: #e9e4f1;
  padding: 10px;
  width: 10%;
  height: 10%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 400ms ease-in border:1px solid #cdcdcd;
  transition: all 400ms ease-in border:1px solid #cdcdcd;
  margin: 0 -1px -1px 0;
  -webkit-transition: all 400ms ease-in;
  transition: all 400ms ease-in;
}

.c0 > span {
  z-index: 5;
}

.c0 > svg {
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  width: 75%;
  height: 75%;
  position: absolute;
  fill: white;
}

<div
  className="c0"
  data-cy="box__selected"
  selected={true}
>
  <svg
    height="100%"
    id="cross-icon"
    viewBox="0 0 24 24"
    width="100%"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <path
      d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
      fill="#d6cde4"
    />
  </svg>
  <span>
    25
  </span>
</div>
`;

exports[`<NumberBox /> it renders when not selected 1`] = `
.c0 {
  position: relative;
  color: #6C4796;
  font-weight: 700;
  background: #ffffff;
  padding: 10px;
  width: 10%;
  height: 10%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 400ms ease-in border:1px solid #cdcdcd;
  transition: all 400ms ease-in border:1px solid #cdcdcd;
  margin: 0 -1px -1px 0;
  -webkit-transition: all 400ms ease-in;
  transition: all 400ms ease-in;
}

.c0 > span {
  z-index: 5;
}

.c0 > svg {
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  width: 75%;
  height: 75%;
  position: absolute;
  fill: white;
}

<div
  className="c0"
  data-cy="box"
>
  <span>
    25
  </span>
</div>
`;
